chainstate/storage/src/internal/version.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[repr(u32)]
+pub enum ChainstateStorageVersion {
+    // skipping 0 for historical reasons
+    V1 = 1,
+}
```

Comment:
Would it make sense to make this implement `Encode`, `Decode` and have the strongly typed version in the DB schema?

Reply:
just realised that this breaks db compatibility with testnet. @TheQuantumPhysicist what's our approach in such cases? Just revert it for now?

Reply:
> just realised that this breaks db compatibility with testnet. @TheQuantumPhysicist what's our approach in such cases? Just revert it for now?

Why? Does it not encode/decode identically to the existing `u32`?

Reply:
Also the approach @ImplOfAnImpl proposes with making this a newtype over `u32` should work in a backwards compatible way.

Reply:
> just realised that this breaks db compatibility with testnet. @TheQuantumPhysicist what's our approach in such cases? Just revert it for now?

It doesn't matter because the worst thing that can happen is a resync... assuming not everyone is going to do it at the same instance, we're fine.

Reply:
But again, it should decide to wipe the db and resync. It shouldn't crash.

Reply:
In case of an enum it actually crashes trying to decode storage version.
```
     Running `target/debug/node-daemon testnet`
thread 'main' panicked at 'db values to be encoded correctly: Error { cause: None, desc: "Input buffer has still data left after decoding!" }', /Users/heorhii/dev/mintlayer-core/chainstate/storage/src/internal/store_tx.rs:395:1
```

Works fine if struct is used (as in my latest commit)

Reply:
We shouldn't let it crash. If there's an incompatibility, we should wipe the DB and start over. This may be related to #1053